ludwig/features/image_feature.py

```diff
@@ -287,9 +288,37 @@ def _finalize_preprocessing_parameters(
             user_specified_num_channels = False
             if preprocessing_parameters[INFER_IMAGE_DIMENSIONS]:
                 user_specified_num_channels = True
-                # Use the maximum num_channels found across all sampled images. torchvision has built-in support for
-                # upsampling images.
-                num_channels = max(num_channels_in_image(x) for x in inferred_sample)
+                channel_frequency = Counter([num_channels_in_image(x) for x in inferred_sample])
```

Comment:
This control flow is already happening in a rather nested chunk of code. Perhaps consider factoring this out into a separate function, i.e. `get_majority_num_channels()`

Reply:
Good idea.  Also did the same for image size inference